From 7ae3981c0e1793129e4e551c1583b2a3db22682d Mon Sep 17 00:00:00 2001
From: jiangfucheng <jiangfucheng0914@foxmail.com>
Date: Thu, 15 Jun 2023 22:57:25 +0800
Subject: [PATCH 1/4] feat(prometheus): allow user configure DEFAULT_BUCKETS

---
 apisix/plugins/prometheus/exporter.lua |  7 ++-
 1 files changed, 7 insertions(+), 1 deletion(-)

diff --git a/apisix/plugins/prometheus/exporter.lua b/apisix/plugins/prometheus/exporter.lua
index 1cb4a534cb9f..8e90326409bc 100644
--- a/apisix/plugins/prometheus/exporter.lua
+++ b/apisix/plugins/prometheus/exporter.lua
@@ -168,10 +168,15 @@ function _M.http_init(prometheus_enabled_in_stream)
             {"code", "route", "matched_uri", "matched_host", "service", "consumer", "node",
             unpack(extra_labels("http_status"))})
 
+    local buckets = DEFAULT_BUCKETS
+    if attr and attr.default_buckets then
+        buckets = attr.default_buckets
+    end
+
     metrics.latency = prometheus:histogram("http_latency",
         "HTTP request latency in milliseconds per service in APISIX",
         {"type", "route", "service", "consumer", "node", unpack(extra_labels("http_latency"))},
-        DEFAULT_BUCKETS)
+        buckets)
 
     metrics.bandwidth = prometheus:counter("bandwidth",
             "Total bandwidth in bytes consumed per service in APISIX",
