diff --git a/apisix/plugins/prometheus/exporter.lua b/apisix/plugins/prometheus/exporter.lua
index d34ab873..dd9d446b 100644
--- a/apisix/plugins/prometheus/exporter.lua
+++ b/apisix/plugins/prometheus/exporter.lua
@@ -235,6 +235,12 @@ end
 
 
 function _M.http_log(conf, ctx)
+    local attr = plugin.plugin_attr("prometheus")
+    local official
+    if attr and attr.official then
+        official = attr.official
+    end
+
     local vars = ctx.var
 
     local route_id = ""
@@ -262,11 +268,17 @@ function _M.http_log(conf, ctx)
         matched_host = ctx.curr_req_matched._host or ""
     end
 
+    if official and official.enable_status then
+
     metrics.status:inc(1,
         gen_arr(vars.status, route_id, matched_uri, matched_host,
                 service_id, consumer_name, balancer_ip,
                 unpack(extra_labels("http_status", ctx))))
 
+    end
+
+    if official and official.enable_latency then
+
     local latency, upstream_latency, apisix_latency = latency_details(ctx)
     local latency_extra_label_values = extra_labels("http_latency", ctx)
 
@@ -284,6 +296,10 @@ function _M.http_log(conf, ctx)
         gen_arr("apisix", route_id, service_id, consumer_name, balancer_ip,
         unpack(latency_extra_label_values)))
 
+    end
+
+    if official and official.enable_bandwidth then
+
     local bandwidth_extra_label_values = extra_labels("bandwidth", ctx)
 
     metrics.bandwidth:inc(vars.request_length,
@@ -293,6 +309,8 @@ function _M.http_log(conf, ctx)
     metrics.bandwidth:inc(vars.bytes_sent,
         gen_arr("egress", route_id, service_id, consumer_name, balancer_ip,
         unpack(bandwidth_extra_label_values)))
+
+    end
 end
 
 
