diff --git a/apisix/utils/upstream.lua b/apisix/utils/upstream.lua
index ece23776..42730ab5 100644
--- a/apisix/utils/upstream.lua
+++ b/apisix/utils/upstream.lua
@@ -111,6 +111,12 @@ local function parse_domain_for_nodes(nodes)
         end
     end
     span:finish(ngx.ctx)
+    if #new_nodes == 0 then
+        local err = "no valid ip found"
+        core.log.error("parse domain for nodes: ", core.json.delay_encode(nodes), " error: ", err)
+        return nil, err
+    end
+
     return new_nodes
 end
 _M.parse_domain_for_nodes = parse_domain_for_nodes
