diff --git a/apisix/utils/upstream.lua b/apisix/utils/upstream.lua
index c39d4cce..0e51ac03 100644
--- a/apisix/utils/upstream.lua
+++ b/apisix/utils/upstream.lua
@@ -79,9 +79,35 @@ local function parse_domain_for_nodes(nodes)
                 core.log.error("dns resolver domain: ", host, " error: ", err)
             end
         else
-            core.table.insert(new_nodes, node)
+            -- patch for: https://github.com/apache/apisix/issues/12436
+            if node.domain then
+                local ip, err = core.resolver.parse_domain(node.domain)
+                if ip then
+                    local new_node = core.table.clone(node)
+                    -- only reset the host to ip here
+                    new_node.host = ip
+                    core.table.insert(new_nodes, new_node)
+                end
+
+                if err then
+                    -- add the old ip back
+                    core.table.insert(new_nodes, node)
+                end
+            else
+                -- add the old ip back
+                core.table.insert(new_nodes, node)
+            end
+
         end
     end
+
+    -- patch for: https://github.com/apache/apisix/issues/10093#issuecomment-1738381865
+    if #new_nodes == 0 then
+        local err = "no valid ip found"
+        core.log.error("parse domain for nodes: ", core.json.delay_encode(nodes), " error: ", err)
+        return nil, err
+    end
+
     return new_nodes
 end
 _M.parse_domain_for_nodes = parse_domain_for_nodes
