diff --git a/apisix/init.lua b/apisix/init.lua
index 548c83b5..8943a2db 100644
--- a/apisix/init.lua
+++ b/apisix/init.lua
@@ -608,6 +608,16 @@ function _M.handle_upstream(api_ctx, route, enable_websocket)
             route, err = parse_domain_in_route(route)
             if err then
                 core.log.error("failed to get resolved route: ", err)
+                if err == "no valid ip found" then
+                    return core.response.exit(502, core.json.encode({
+                        data = core.json.null,
+                        result = false,
+                        code_name = "BAD_GATEWAY",
+                        code = 1650200,
+                        message = "502 Bad Gateway: parse backend host fail, no valid ip found",
+                    }))
+                end
+
                 return core.response.exit(500)
             end
 
