diff --git a/apisix/plugins/opentelemetry.lua b/apisix/plugins/opentelemetry.lua
index f8013e6f..f9d2f3ca 100644
--- a/apisix/plugins/opentelemetry.lua
+++ b/apisix/plugins/opentelemetry.lua
@@ -329,7 +329,7 @@ function _M.rewrite(conf, api_ctx)
     end
 
     local ctx = tracer:start(upstream_context, api_ctx.var.request_uri, {
-        kind = span_kind.server,
+        kind = span_kind.client,
         attributes = attributes,
     })
     api_ctx.otel_context_token = ctx:attach()
